File: src/main/java/dev/esophose/discordbot/command/arguments/InetAddressArgumentHandler.java
package dev.esophose.discordbot.command.arguments;

import dev.esophose.discordbot.command.DiscordCommandArgumentHandler;
import discord4j.core.object.entity.Guild;
import java.net.InetAddress;
import reactor.core.publisher.Mono;

public class InetAddressArgumentHandler extends DiscordCommandArgumentHandler<InetAddress> {

    @Override
    public Mono<InetAddress> handleInternal(Guild guild, String input) {
        if (input.trim().isEmpty())
            return Mono.empty();

        try {
            return Mono.just(InetAddress.getByName(input));
        } catch (Exception ex) {
            return Mono.empty();
        }
    }

    @Override
    public String getErrorMessage(Guild guild, String input) {
        return "Unknown host: [" + input + "]";
    }

    @Override
    public Class<InetAddress> getHandledType() {
        return InetAddress.class;
    }

}

File: src/main/kotlin/dev/esophose/discordbot/command/arguments/InetAddressArgumentHandler.kt
package dev.esophose.discordbot.command.arguments

import dev.esophose.discordbot.command.DiscordCommandArgumentHandler
import discord4j.core.`object`.entity.Guild
import reactor.core.publisher.Mono
import java.net.InetAddress
import kotlin.reflect.KClass

class InetAddressArgumentHandler : DiscordCommandArgumentHandler<InetAddress>() {

    override val handledType: KClass<InetAddress>
        get() = InetAddress::class

    public override fun handleInternal(guild: Guild, input: String): Mono<InetAddress> {
        if (input.trim { it <= ' ' }.isEmpty())
            return Mono.empty()

        return try {
            Mono.just(InetAddress.getByName(input))
        } catch (ex: Exception) {
            Mono.empty()
        }
    }

    override fun getErrorMessage(guild: Guild, input: String): String {
        return "Unknown host: [$input]"
    }

}
